Header header
string state
string command